src/chessboard/types/index.ts

```diff
@@ -252,6 +252,14 @@ export type ChessboardProps = {
    * @default () => {}
    */
   onArrowsChange?: (squares: Square[][]) => void;
+  /**
+   * User function that is run when piece is dropped. Must return whether the move results in a promotion or not.
+   * @default () => false
+   */
+  onCheckForPromotion?: (
```

Comment:
I guess for this function, you have to call  [chess.moves({ square: sourceSquare })](https://github.com/jhlywa/chess.js/blob/master/README.md#moves-piece-piece-square-square-verbose-boolean-) then check if your `targetSquare` included in the returned list (?)

So I would like to suggest instead implement more universal user function, which  accepts only `sourceSquare` and returns all possible moves from that square. 

```suggestion
  getValidMovesFromSquare?: (square: Square) => Square[]
```

then you can check promotion in `handlePromotion` like this:

```
  if(getValidMovesFromSquare(sourceSquare).includes(targetSquare)){ // do promotion }
``` 

This function can be useful in the future as well, I think, What is your opinion guys? 
@Clariity @MatthewHerrmannDev

Reply:
I prefer the onCheckForPromotion approach

It allows for people to do what they want better, with less logic applied within the package